### PR TITLE
Fixes for Windows and add CI workflow

### PR DESCRIPTION
**Pull request by LuKks**

### CODE REVIEW

File: test/del.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ test('del(key) folder', async function (t) {
     await drive.del('/examples')
     t.fail('should have given error')
   } catch (error) {
-    t.is(error.code, 'EISDIR')
+    t.ok(['EISDIR', 'EPERM'].indexOf(error.code) !== -1)
```
Comment (mafintosh) [inline test/del.js:30]: just use an ||

**Comment by LuKks** [comment]
Maybe we can handle LICENSE.shortcut test file better now that I know about Windows but it's also okay as it is.

File: test/entry.js [hunk 2]
```diff
@@ -87,7 +87,7 @@ test('entry(key) permission denied', async function (t) {
   // + should we ignore permission errors and just return null?
   try {
     await drive.get('/key.secret')
-    t.fail('should have given error')
+    if (!isWin) t.fail('should have given error')
```
Comment (LuKks) [inline test/entry.js:90]: Is there a way to make the read permission fail on Windows?

File: test/create-write-stream.js [hunk 3]
```diff
@@ -30,7 +30,7 @@ test('createWriteStream(key) with options', async function (t) {
   t.alike(await drive.entry('/new-script.sh'), {
     key: '/new-script.sh',
     value: {
-      executable: true,
+      executable: !isWin,
```
Comment (LuKks) [inline test/create-write-stream.js:33]: I decided to not skip the entire test in this case, so at least createWriteStream() with options is run anyway.

File: test/create-write-stream.js [hunk 4]
```diff
@@ -77,7 +77,7 @@ test('createWriteStream(key) replace', async function (t) {
   t.alike(await drive.get('/LICENSE'), buffer)
 })
 
-test('createWriteStream(key) replace existing executable', async function (t) {
+test('createWriteStream(key) replace existing executable', { skip: isWin }, async function (t) {
```
Comment (LuKks) [inline test/create-write-stream.js:80]: This would be okay to skip it as there is another test replacing a normal file.

**Comment by LuKks** [comment]
@mafintosh Ready!

File: index.js [hunk 5]
```diff
@@ -30,14 +30,15 @@ module.exports = class Localdrive {
     }
 
     if (stat.isSymbolicLink()) {
-      entry.value.linkname = (await fsp.readlink(filename)).slice(this.root.length)
+      const link = (await fsp.readlink(filename)).slice(this.root.length)
+      entry.value.linkname = unixPathResolve('/', link)
```
Comment (mafintosh) [inline index.js:34]: ah, we should support relative paths here

Comment (mafintosh) [reply index.js:34]: ie replace the \ with /, that’s it

Comment (LuKks) [reply index.js:34]: Should I still remove the root from the `linkname`?
About resolving the link, it's because I treated `linkname` as any other key

**Pull request merged**